halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/services/v1/Daemon.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.cli.services.v1;
+
+import retrofit.RestAdapter;
+import retrofit.RetrofitError;
+import retrofit.client.OkClient;
+
+import java.net.ConnectException;
+
+public class Daemon {
+  public static DaemonService getService() {
+    return getService(false);
+  }
+
+  public static DaemonService getService(boolean log) {
```

Comment:
I do this kind of thing too, but want to point out that log only matters if it needs to create the service. Since you also have a getService() without log, it looks suspicious. This is a singleton and log is global, so perhaps you should set the log preference as global state and get rid of this method. That way the individual call sites dont need to replicate responsibility over figuring out whether they should log or not and a new site wont introduce a bug if it is using different rules.

Have one place determine the logging policy and set it here. It isnt the consumer's problem to worry about unless you have a logging and non-logging variant and they get what they ask for.